muacryptcc/plugin.py

```diff
@@ -6,41 +6,65 @@
 from hippiehug import Chain
 from claimchain import State, View
 from claimchain.crypto.params import LocalParams
-from claimchain.utils import pet2ascii
+from claimchain.utils import pet2ascii, ascii2pet
+from .filestore import FileStore
 
 hookimpl = pluggy.HookimplMarker("muacrypt")
 
 
 @hookimpl
 def instantiate_account(plugin_manager, basedir):
     cc_dir = os.path.join(basedir, "muacryptcc")
-    cc_manager = CCAccount(cc_dir)
+    store_dir = os.path.join(cc_dir, "store")
+    store = FileStore(store_dir)
+    cc_manager = CCAccount(cc_dir, store)
     plugin_manager.register(cc_manager)
 
 
-class CCAccount:
+class CCAccount(object):
     def __init__(self, accountdir, store=None):
         self.accountdir = accountdir
         if not os.path.exists(accountdir):
             os.makedirs(accountdir)
         self.store = store
         self.init_crypto_identity()
 
+    #
+    # muacrypt plugin hook implementations
+    #
+    @hookimpl
+    def process_incoming_gossip(self, addr2pagh, account_key, dec_msg):
+        root_hash = dec_msg["GossipClaims"]
+        store = FileStore(dec_msg["ChainStore"])
+        peers_chain = Chain(store, root_hash=ascii2pet(root_hash))
+        assert peers_chain
+        peers_pk = View(peers_chain).params.dh.pk
+        assert peers_pk
+
+    @hookimpl
+    def process_outgoing_before_encryption(self, account_key, msg):
+        msg["GossipClaims"] = pet2ascii(self.head)
+        assert self.store.url
```

Comment:
this assert makes no sense to me. we don't need to assert we have a "valid" store or so here.